src/watertap_contrib/reflo/analysis/multiperiod/ltmed_vagmd_semibatch/MED_VAGMD_semibatch_class.py

```diff
@@ -0,0 +1,767 @@
+    get_scaling_factor,
+)
+
+from idaes.core import UnitModelCostingBlock
+import idaes.logger as idaeslog
+from idaes.core.solvers.get_solver import get_solver
+
+# WaterTAP imports
+from watertap_contrib.reflo.costing import TreatmentCosting
+
+# Flowsheet function imports
+from watertap_contrib.reflo.analysis.multiperiod.ltmed_vagmd_semibatch.MED_VAGMD_flowsheet import (
+    build_med_md_flowsheet,
+    fix_dof_and_initialize,
+)
+
+__author__ = "Zhuoran Zhang"
+
+solver = get_solver()
+
+
+def get_variable_pairs(t1, t2):
+    """
+    This function returns paris of variables that need to be connected across two time periods
+    Args:
+        t1: current time block
+        t2: next time block
+    Returns:
+        None
+    """
+    return [
+        # Take MD feed properties from last step mixer
+        (t1.fs.S2.MD_feed_state[0].temperature, t2.fs.vagmd.feed_props[0].temperature),
+        (t1.fs.S2.MD_feed_state[0].pressure, t2.fs.vagmd.feed_props[0].pressure),
+        (
+            t1.fs.S2.MD_feed_state[0].flow_mass_phase_comp["Liq", "TDS"],
+            t2.fs.vagmd.feed_props[0].flow_mass_phase_comp["Liq", "TDS"],
+        ),
+        (
+            t1.fs.S2.MD_feed_state[0].flow_mass_phase_comp["Liq", "H2O"],
+            t2.fs.vagmd.feed_props[0].flow_mass_phase_comp["Liq", "H2O"],
+        ),
+        # Inherit mixer properties from last step
+        (
+            t1.fs.S2.remained_liquid_state[0].temperature,
+            t2.fs.M1.remained_liquid_state[0].temperature,
+        ),
+        (
+            t1.fs.S2.remained_liquid_state[0].pressure,
+            t2.fs.M1.remained_liquid_state[0].pressure,
+        ),
+        (
+            t1.fs.S2.remained_liquid_state[0].flow_mass_phase_comp["Liq", "TDS"],
+            t2.fs.M1.remained_liquid_state[0].flow_mass_phase_comp["Liq", "TDS"],
+        ),
+        (
+            t1.fs.S2.remained_liquid_state[0].flow_mass_phase_comp["Liq", "H2O"],
+            t2.fs.M1.remained_liquid_state[0].flow_mass_phase_comp["Liq", "H2O"],
+        ),
+        # Mixer volume
+        (t1.fs.volume_in_tank, t2.fs.volume_in_tank_previous),
+    ]
+
+
+def unfix_dof(m):
+    """
+    This function unfixes a few degrees of freedom for optimization
+    Args:
+        m: object containing the integrated nuclear plant flowsheet
+    Returns:
+        None
+    """
+
+    m.fs.vagmd.feed_props[0].temperature.unfix()
+    m.fs.vagmd.feed_props[0].pressure.unfix()
+    m.fs.vagmd.feed_props[0].flow_mass_phase_comp["Liq", "H2O"].unfix()
+    m.fs.vagmd.feed_props[0].flow_mass_phase_comp["Liq", "TDS"].unfix()
+
+    m.fs.M1.remained_liquid_state[0].temperature.unfix()
+    m.fs.M1.remained_liquid_state[0].pressure.unfix()
+    m.fs.M1.remained_liquid_state[0].flow_mass_phase_comp["Liq", "H2O"].unfix()
+    m.fs.M1.remained_liquid_state[0].flow_mass_phase_comp["Liq", "TDS"].unfix()
+
+    m.fs.volume_in_tank_previous.unfix()
+
+    return
+
+
+@declare_process_block_class("MEDVAGMDsemibatch")
```

Comment:
@adam-a-a this looks like a unit model to me (which is totally fine) but wondering if we should move it to the unit model folder?

Reply:
I guess I am still contemplating why @Zhuoran29 chose to formulate the model this way. Can you elaborate on why you found this way to be the better choice for implementation rather than the "typical approach" for creating the multiperiod model? 

Also, even though this is formulated like a unit model, I would lean towards a new, higher-level folder that comprises sub-flowsheets of interest that could hypothetically link to other unit models and similar subflowsheets, if we want to keep the formulation this way.

Reply:
I basically inherit the structure of VAGMD batch model, which was built as a unit model because I thought it'd be easier to integrate in a larger flowsheet and to avoid mixing with a higher-level multiperiod structure (for example in scenarios where we want to do hourly/daily simulation with varied energy/feed and would want to keep the model performance steady-state during that hour/day).

But now I think the "typical approach" should be fine as well.